crates/orderbook/openapi.yml

```
@@ -375,6 +375,22 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/VersionResponse"
+  /api/v1/app_data/by_hash/{app_data_hash}:
```

API naming nit:

```suggestion
  /api/v1/app-data/{app_data_hash}:
```

I think "by hash" is superfluous (as it is obviously the way to identify app-hashes).

---

Typically, in "REST" terms (although we are very loosely REST-y) you typically organize routes as `GET resource/${id}/sub-resouces...`. So, for fetching "app-data", you would identify it by its ID (for which only hashes really make sense IMO).

If you were, say, to fetch APP data by order ID, you could organize it like: `GET order/${orderUid}/app-data`. Or, another example, fetch all app-datas by transaction: `GET transaction/${transactionHash}/app-datas`, etc.

For example, I'm also not a fan of the `solver_competitions/...` routes either, and would probably have done it like:
```
GET auction/${auctionId}/solver_competition
GET transaction/${txHash}/solver_competition
```

Regarding the kebab case vs snake case, all our other routes use snake case so we should keep it that way for consistency. (so disregard my suggestion of `app_data` -> `app-data`).

None of this is important and has no technical reason behind it, like I said, its just bike-shedding and nit-picking (I approved the PR as is).

---

changed